File: Bye World/byeworld-frontend/src/model/Listing.ts
import { City } from "./City";
import { Company } from "./Company";
import { RequiresSkill } from "./RequiresSkill";

export interface Listing {
  id: number;
  title: string;
  description: string;
  requirements: RequiresSkill[];
  postingDate: Date;
  closingDate: Date;
  company?: Company;
  city?: City;
}

File: Bye World/ByeWorld-backend/Models/Listing.cs
namespace ByeWorld_backend.Models
{
    public class Listing
    {
        public long Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public List<RequiresSkill> Requirements { get; set; } = new List<RequiresSkill>();
        public DateTime PostingDate { get; set; } = DateTime.UtcNow;
        public DateTime ClosingDate { get; set; } = DateTime.MaxValue;
        public Company? Company{ get; set; }
        public City? City { get; set; }
    }
}
